packages/react/src/Bento/Bento.tsx

```diff
@@ -0,0 +1,298 @@
+} & React.HTMLAttributes<HTMLDivElement> &
+  BaseProps<HTMLDivElement>
+
+const returnClassBasedOnResponsiveMap = (
+  classIdentifier: string,
+  propName: string,
+  prop?:
+    | ResponsiveColumnIndex
+    | ColumnIndex
+    | ResponsiveRowIndex
+    | number
+    | ResponsiveFlow
+    | Flow
+    | ResponsiveAlign
+    | Align
+    | ResponsivePadding
+    | Padding,
+) => {
+  const classesToMerge: string[] = []
+  if (typeof prop === 'number' || typeof prop === 'string') {
+    classesToMerge.push(styles[`${classIdentifier}--${propName}-${prop}`])
+  } else if (typeof prop === 'object') {
+    for (const [key, value] of Object.entries(prop)) {
+      classesToMerge.push(styles[`${classIdentifier}--${key}-${propName}-${value}`])
+    }
+  }
+  return classesToMerge
+}
+
+type ReturnOrderedChildrenProps = {
+  styleOrder: StyleOrder | ResponsiveStyleOrder
+  children: React.ReactNode
+  currentBreakpointSize?: BreakpointSize
+}
+
+const returnOrderedChildren = ({styleOrder, children, currentBreakpointSize}: ReturnOrderedChildrenProps) => {
+  if (typeof styleOrder === 'string') {
+    return styleOrder === 'default' ? children : React.Children.toArray(children).reverse()
+  } else if (typeof styleOrder === 'object' && currentBreakpointSize) {
+    // This handles setting the styleOrder based on breakpoints as you might expect via CSS
+    const filledStyleOrder: ResponsiveStyleOrder = {}
+    let previousSize: StyleOrder = 'default'
+    for (const size of Object.keys(BreakpointSize)) {
+      const lowerCaseSize = size.toLocaleLowerCase()
+      if (styleOrder[lowerCaseSize]) previousSize = styleOrder[lowerCaseSize]
+      filledStyleOrder[lowerCaseSize] = styleOrder[lowerCaseSize] || previousSize
+    }
+    return filledStyleOrder[currentBreakpointSize] === 'default' ? children : React.Children.toArray(children).reverse()
+  } else {
+    return children
+  }
+}
+
+const Item = ({
+  className,
+  columnStart,
+  columnSpan,
+  rowStart,
+  rowSpan,
+  flow = 'row',
+  colorMode,
+  visualAsBackground = false,
+  styleOrder = 'default',
+  children,
+  ...rest
+}: BentoItemProps) => {
+  const {currentBreakpointSize} = useWindowSize()
+  const bentoItemClassArray = [styles.Bento__Item]
+  bentoItemClassArray.push(
+    ...returnClassBasedOnResponsiveMap('Bento__Item', 'column-span', columnSpan),
+    ...returnClassBasedOnResponsiveMap('Bento__Item', 'row-span', rowSpan),
+    ...returnClassBasedOnResponsiveMap('Bento__Item', 'column-start', columnStart),
+    ...returnClassBasedOnResponsiveMap('Bento__Item', 'row-start', rowStart),
+    ...returnClassBasedOnResponsiveMap('Bento__Item', 'flow', flow),
+  )
+
+  if (!visualAsBackground) {
+    flow === 'column' && bentoItemClassArray.push(styles['Bento-column-padding-override'])
+    flow === 'row' && bentoItemClassArray.push(styles['Bento-row-padding-override'])
+  }
+
+  const colorModeProp = colorMode ? {'data-color-mode': colorMode} : {}
```

Comment:
I asked this before, but the conversation thread seems to have gone 🤔 so asking again 😅 ... why do we need to set data-color-mode manually instead of using ThemeProvider, which [supports nesting](https://primer-f4d7c33f83-26139705.drafts.github.io/storybook/?path=/story/components-themeprovider--nested) and can apply the same attribute in a centralised location?

Reply:
Hey Reza, I answered this here: https://github.com/primer/brand/pull/390#issuecomment-1701107528